### ISSUE TITLE
Connecting to a server displays blocks incorrectly

### ISSUE THREAD

**Issue by Jack-Papel** [devs-immortal/The-Aether#112]
When connecting to the hypixel bedwars server using this mod, I was able to load in, but many things rendered incorrectly. The most noticeable was that many blocks displayed as stairs in various orientations. I can't provide much more info because this kind of glitch is very scary to me, for some reason. Once experiencing, I freaked out and closed the game.

**Comment by MBatt1** [comment]
Does this happen on all servers? or just hypixel?

**Comment by Jack-Papel** [comment]
The issue also occurs on mineplex and minehut. Attached are screenshots of what it looked like on mineplex. (Minehut crashed my game, so there are no screenshots.)
It seems the issue isn't just stairs, as I had thought. The block id's seem to be randomized or maybe offset. (Blocks that would be stone, for example, would all be entirely another block.)

![2021-02-25_13 30 12](https://user-images.githubusercontent.com/64770632/109220438-c7133f00-776d-11eb-9db1-ff092137ba71.png)
![2021-02-25_13 30 14](https://user-images.githubusercontent.com/64770632/109220439-c7abd580-776d-11eb-86e1-910638260aa6.png)
![2021-02-25_13 30 17](https://user-images.githubusercontent.com/64770632/109220440-c7abd580-776d-11eb-8d2e-574595e30330.png)
![2021-02-25_13 30 19](https://user-images.githubusercontent.com/64770632/109220443-c7abd580-776d-11eb-9dec-ce7fc9cb7b8c.png)
![2021-02-25_13 30 10](https://user-images.githubusercontent.com/64770632/109220448-ca0e2f80-776d-11eb-8418-a61b92b91f2e.png)

(I've since been banned from hypixel, for unrelated reasons, so I can't show screenshots of that server.)

**Comment by MBatt1** [comment]
I'm unsure of the cause specifically because I'm more on the art side of things, but I would have to guess that this is a result of the plugins that allow for multiple versions to connect to these servers, and the old id system (pre-flattening) conflicting with something in fabric or in the aether mod.

**Comment by Jack-Papel** [comment]
I tested it on a server that only works on 1.16.4 (mc.fruitservers.net). The issue still arose. I made sure to test joining the server on 1.12.2 to make sure, and the server only supports the latest version.
I also tested joining with only geckolib and fabric api, and it worked fine.

**Comment by Jack-Papel** [comment]
I have done more testing on a LAN server. (The host was a demo account, if that matters)
I took more screenshots but unfortunately Windows sucks so I only have this one.

![image](https://user-images.githubusercontent.com/64770632/110883140-d9c26380-8297-11eb-8f16-0a4e016e6e6f.png)

If the host and the client are using the aether mod, it works fine. If the host isn't, (here they're using optifine 1.16.4), then it doesn't work.

Notes:
The observed block is the one that collision works with (A.K.A. if grass -> bed, it'd use the bed collision)
If you place a block (here for example, piston), it'll place the wrong one (here, a green bed head)
Middle click selects the observed block, not the actual block.
Everything looks normal in the inventory
Mobs seem to be the same.
Tile Entities are also affected

Here's a list of the differences I noticed:
Sticky Piston (All orientations) -> green bed head
Farmland -> redstone
Grass -> Invisible Green bed head
Tall grass -> Waterlogged Dark Prismarine stairs
Cave air -> Waterlogged Dead Horn coral thing
Path -> Rotated purpur stair
Nether Portal -> cactus
Obsidian -> Piston head
Brown Banner -> Red Banner
Some beds map to the wrong bed color, often with a rotated bed head, while others map to noteblocks.
Sandstone -> Leaves
Chests -> Spruce Stairs (But are still openable)
Lilypads -> Stone Brick Stairs
Clay -> Oak Pressure Plate
When I was in a swamp, some water blocks were gone (Not most)
Oak Pressure Plate -> Lever
Snow (Layer) -> Iron door
Orange Terracotta -> Quartz Stairs
Flowers are also all wrong
Appears that acacia stairs map to different colors of stained glass
Mycelium -> Top Stone Brick Slab
Mushroom block -> Mushroom block (but the wrong kind)

Things that stay the same:
Grass, stone, water, air, dirt, stone variants, logs, leaves, etc.
Basically, most of the blocks that aren't on the front page of the creative menu are messed up. 

My best guess is that some block ID's are offset, which would explain the different orientations of acacia stair corresponding to different stained glass colors, as well as common blocks (lower block ID #) being unaffected.

I'd be happy to provide more details if needed.

**Comment by Haven-King** [comment]
This is a result of registry sync not working. Registry sync is a feature that Fabric API uses to make sure that blocks on the client and server are registered to the same internal number ID's. If the server has mods with blocks that the client doesn't have and the client doesn't have FAPI, or vice versa, this is what happens. This is not an issue with the Aether specifically, rather, this same thing would most likely happen with any other content mods.
